### ISSUE TITLE
Interactive Slider Issue: 'Figure' object has no attribute 'canvas'

### ISSUE THREAD

**Issue by PrParadox** [pyodide/matplotlib-pyodide#51]
The following code is taken directly from matplotlib website https://matplotlib.org/stable/gallery/widgets/slider_demo.html. It generates red errors 'Figure' object has no attribute 'canvas'. Even though the plot itself work as expected! I tried to put everything in a try block, didn't help. Is there anyway the catch this error?  

```
Uncaught PythonError: Traceback (most recent call last):
  File "/lib/python3.11/site-packages/matplotlib_pyodide/browser_backend.py", line 221, in onmousemove
    self.motion_notify_event(x, y, guiEvent=event)
  File "/lib/python3.11/site-packages/matplotlib/backend_bases.py", line 1926, in motion_notify_event
    self.callbacks.process(s, event)
  File "/lib/python3.11/site-packages/matplotlib/cbook/__init__.py", line 292, in process
    self.exception_handler(exc)
  File "/lib/python3.11/site-packages/matplotlib/cbook/__init__.py", line 96, in _exception_printer
    raise exc
  File "/lib/python3.11/site-packages/matplotlib/cbook/__init__.py", line 287, in process
    func(*args, **kwargs)
  File "/lib/python3.11/site-packages/matplotlib/widgets.py", line 534, in _update
    self.set_val(val)
  File "/lib/python3.11/site-packages/matplotlib/widgets.py", line 568, in set_val
    self._observers.process('changed', val)
  File "/lib/python3.11/site-packages/matplotlib/cbook/__init__.py", line 292, in process
    self.exception_handler(exc)
  File "/lib/python3.11/site-packages/matplotlib/cbook/__init__.py", line 96, in _exception_printer
    raise exc
  File "/lib/python3.11/site-packages/matplotlib/cbook/__init__.py", line 287, in process
    func(*args, **kwargs)
  File "/lib/python3.11/site-packages/matplotlib/widgets.py", line 585, in <lambda>
    return self._observers.connect('changed', lambda val: func(val))
                                                          ^^^^^^^^^
  File "<exec>", line 92, in update
AttributeError: 'Figure' object has no attribute 'canvas'
```

```python
import matplotlib.pyplot as plt
import numpy as np

from matplotlib.widgets import Button, Slider


# The parametrized function to be plotted
def f(t, amplitude, frequency):
    return amplitude * np.sin(2 * np.pi * frequency * t)

t = np.linspace(0, 1, 1000)

# Define initial parameters
init_amplitude = 5
init_frequency = 3

# Create the figure and the line that we will manipulate
fig, ax = plt.subplots()
line, = ax.plot(t, f(t, init_amplitude, init_frequency), lw=2)
ax.set_xlabel('Time [s]')

# adjust the main plot to make room for the sliders
fig.subplots_adjust(left=0.25, bottom=0.25)

# Make a horizontal slider to control the frequency.
axfreq = fig.add_axes([0.25, 0.1, 0.65, 0.03])
freq_slider = Slider(
    ax=axfreq,
    label='Frequency [Hz]',
    valmin=0.1,
    valmax=30,
    valinit=init_frequency,
)

# Make a vertically oriented slider to control the amplitude
axamp = fig.add_axes([0.1, 0.25, 0.0225, 0.63])
amp_slider = Slider(
    ax=axamp,
    label="Amplitude",
    valmin=0,
    valmax=10,
    valinit=init_amplitude,
    orientation="vertical"
)


# The function to be called anytime a slider's value changes
def update(val):
    line.set_ydata(f(t, amp_slider.val, freq_slider.val))
    fig.canvas.draw_idle()


# register the update function with each slider
freq_slider.on_changed(update)
amp_slider.on_changed(update)

# Create a `matplotlib.widgets.Button` to reset the sliders to initial values.
resetax = fig.add_axes([0.8, 0.025, 0.1, 0.04])
button = Button(resetax, 'Reset', hovercolor='0.975')


def reset(event):
    freq_slider.reset()
    amp_slider.reset()
button.on_clicked(reset)

plt.show()
```

**Comment by hoodmane** [comment]
What about:
```
def update(val):
    try:
        line.set_ydata(f(t, amp_slider.val, freq_slider.val))
        fig.canvas.draw_idle()
    except Exception:
       pass
```

**Comment by PrParadox** [comment]
@hoodmane try block couldn't fix this no matter how hard I tried. So I ended up making a minimal working example to pinpoint where the problem is. I was using Pyscript (which uses Pyodide under-hood) for the ease of use. However, pyscript blocks were not as isolated as I hoped, they do share the same environment, even though I did set different env in their tags. 

The reason I was getting this error was that some other fig in another separate tag was somehow gracefully stealing the slider and causing errors. What is weird though, is that both figs were working completely fine, well, except for generating obscure errors. At any rate, thanks.
